### PR TITLE
SDIT-1522 Location From Nomis sync

### PR DESCRIPTION
**Pull request by steverendell**
Initial PR using pattern copied from incidents and NAs

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.locations
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.swagger.v3.oas.annotations.media.Schema
+import java.time.LocalDate
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(description = "Filter specifying what should be migrated from NOMIS to the Non-Associations service")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt:8]: Copy and paste of "Non-Associations"

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+data class LocationsMigrationFilter(
+
+  @Schema(
+    description = "Only include non-associations created on or after this date",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt:12]: non-assoc

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    example = "2020-03-23",
+  )
+  val fromDate: LocalDate? = null,
+
+  @Schema(
+    description = "Only include non-associations created before or on this date",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt:18]: non-assoc

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt (hunk 3 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/locations/LocationsMigrationFilter.kt:18]: thanks mike, as it happens ill probably just delete the filter next time as I dont think itll be needed

**Pull request merged**